Cypress workspace test is flaky when switching between views

**Describe the bug**
The workspace test has been flaky recently.

It tries to render a workspace when there is none after switching to the 'Recent' view.

**To Reproduce**
Steps to reproduce the behavior:
See https://github.com/nextcloud/text/actions/runs/16546964854/job/46796529217 for an example of a failed run.

<details>
<summary>Cypress output</summary>

```
   1) Workspace
       Hides the workspace when switching to another view:
     AssertionError: Timed out retrying after 10000ms: Expected <div#rich-workspace> not to exist in the DOM, but it was continuously found.
      at Context.eval (http://localhost:8081/__cypress/tests?p=cypress/e2e/workspace.spec.js:639:33)

```

</details>

<img width="1280" height="720" alt="Image" src="https://github.com/user-attachments/assets/92d7bfe6-0eda-4669-9a89-7a678afd6dd6" />

**Expected behavior**
Workspace should only show if there is one and tests should pass reliably

**Screenshots**
Some debugging revealed that `localHasRichWorkspace` is true despite `hasRichWorkspace` having switched to false. This is most likely due to a `getFileInfo` request from the previous view finishing afterwards.

<img width="1280" height="720" alt="Image" src="https://github.com/user-attachments/assets/a7fb4f7d-f805-44e7-949c-30192c36993a" />


**Server details:**
Text github CI

--- a/src/helpers/files.js
+++ b/src/helpers/files.js
@@ -192,13 +192,15 @@ let FilesHeaderRichWorkspaceView
 let FilesHeaderRichWorkspaceInstance
 let latestFolder
 
+function enabled(_, view) {
+	return ['files', 'favorites', 'public-share'].includes(view.id)
+}
+
 export const FilesWorkspaceHeader = new Header({
 	id: 'workspace',
 	order: 10,
+	enabled,
 
-	enabled(_, view) {
-		return ['files', 'favorites', 'public-share'].includes(view.id)
-	},
 	render: async (el, folder) => {
 		latestFolder = folder
 		// Import the RichWorkspace component only when needed
@@ -231,14 +233,15 @@ export const FilesWorkspaceHeader = new Header({
 		window.FilesHeaderRichWorkspaceInstance = FilesHeaderRichWorkspaceInstance
 	},
 
-	updated(folder) {
+	updated(folder, view) {
 		latestFolder = folder
 		if (!FilesHeaderRichWorkspaceInstance) {
 			console.error('No vue instance found for FilesWorkspaceHeader')
 			return
 		}
 
-		const hasRichWorkspace = !!folder.attributes['rich-workspace-file']
+		const hasRichWorkspace =
+			!!folder.attributes['rich-workspace-file'] && enabled(folder, view)
 		FilesHeaderRichWorkspaceInstance.hasRichWorkspace = hasRichWorkspace
 		FilesHeaderRichWorkspaceInstance.content =
 			folder.attributes['rich-workspace'] || ''

--- a/src/views/RichWorkspace.vue
+++ b/src/views/RichWorkspace.vue
@@ -5,7 +5,7 @@
 
 <template>
 	<div
-		v-if="enabled && localHasRichWorkspace"
+		v-if="shouldRender"
 		id="rich-workspace"
 		:class="{ focus: focus, dark: darkTheme }">
 		<RichTextReader
@@ -25,8 +25,7 @@
 			active
 			rich-workspace
 			@ready="ready = true"
-			@focus="onFocus"
-			@error="reset" />
+			@focus="onFocus" />
 	</div>
 </template>
 
@@ -89,6 +88,7 @@ export default {
 			loaded: false,
 			ready: false,
 			autofocus: false,
+			shouldAutofocus: false,
 			hideMenu: true,
 			darkTheme: window?.OCA?.Accessibility?.theme === 'dark',
 			enabled: window?.OCA?.Text?.RichWorkspaceEnabled,
@@ -98,10 +98,16 @@ export default {
 		shareToken() {
 			return getSharingToken()
 		},
+		shouldRender() {
+			return this.enabled && this.localHasRichWorkspace
+		},
 	},
 	watch: {
 		path() {
-			this.getFileInfo()
+			this.reset()
+		},
+		ready() {
+			this.shouldAutofocus = false
 		},
 		focus(newValue) {
 			if (!newValue) {
@@ -110,18 +116,17 @@ export default {
 					.scrollTo(0, 0)
 			}
 		},
-		hasRichWorkspace(value) {
-			this.localHasRichWorkspace = value
+		shouldRender(value) {
 			if (value) {
 				this.getFileInfo()
 			}
 		},
+		hasRichWorkspace(value) {
+			this.localHasRichWorkspace = value
+		},
 	},
 	mounted() {
 		this.localHasRichWorkspace = this.hasRichWorkspace
-		if (this.enabled && this.hasRichWorkspace) {
-			this.getFileInfo()
-		}
 		subscribe('Text::showRichWorkspace', this.showRichWorkspace)
 		subscribe('Text::hideRichWorkspace', this.hideRichWorkspace)
 		subscribe('files:node:created', this.onFileCreated)
@@ -146,22 +151,19 @@ export default {
 			this.unlistenKeydownEvents()
 		},
 		reset() {
-			this.localHasRichWorkspace = false
 			this.file = null
 			this.focus = false
+			this.shouldAutofocus = false
 			this.$nextTick(() => {
-				this.creating = false
-				this.getFileInfo()
+				if (this.shouldRender) {
+					this.getFileInfo()
+				}
 			})
 		},
-		getFileInfo(autofocus) {
-			if (!this.enabled) {
-				return
-			}
+		getFileInfo() {
 			this.file = null
 			this.ready = false
 			this.loaded = true
-			this.autofocus = false
 			const params = { path: this.path }
 			if (IS_PUBLIC) {
 				params.shareToken = this.shareToken
@@ -174,8 +176,7 @@ export default {
 					this.file = data.file
 					this.editing = true
 					this.loaded = true
-					this.autofocus = autofocus || false
-					this.localHasRichWorkspace = true
+					this.autofocus = this.shouldAutofocus
 					return true
 				})
 				.catch((error) => {
@@ -190,13 +191,11 @@ export default {
 					this.file = null
 					this.loaded = true
 					this.ready = true
-					this.creating = false
 					return false
 				})
 		},
 		showRichWorkspace(event) {
 			this.enabled = true
-			this.getFileInfo(event?.autofocus || false)
 		},
 		hideRichWorkspace() {
 			this.enabled = false
@@ -214,8 +213,8 @@ export default {
 		},
 		onFileCreated(node) {
 			if (SUPPORTED_STATIC_FILENAMES.includes(node.basename)) {
+				this.shouldAutofocus = this.enabled
 				this.localHasRichWorkspace = true
-				this.getFileInfo(true)
 			}
 		},
 		onFileDeleted(node) {

